pkg/elevator/internal/webhooks/tenantobj_webhook.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	catalogv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/catalog/v1alpha1"
+	elevatorutil "github.com/kubermatic/kubecarrier/pkg/elevator/internal/util"
+)
+
+// TenantObjWebhookHandler handles validating of TenantObjs.
+type TenantObjWebhookHandler struct {
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	decoder *admission.Decoder
+
+	client.Client
+	NamespacedClient client.Client
+
+	TenantGVK, ProviderGVK schema.GroupVersionKind
+
+	DerivedCRName, ProviderNamespace string
+}
+
+var _ admission.Handler = (*TenantObjWebhookHandler)(nil)
+
+// Handle is the function to handle validating requests of ManagementClusterObjs.
```

Comment:
```suggestion
// Handle is the function to handle ManagementClusterObjs validating requests.
```

Reply:
Also before you're talking about TenantObjects, and now you're talking about ManagementClusterObjs? 

Furthermore, provider/tenant concepts should be orthogonal to the elevator. It's true that the majority of the code is written with that assumption in mind, thus for consistency we should honor that practice.

Reply:
Sorry the comment is copied from the `Catapult` webhook, it should be `TenantObjects`